### PR TITLE
Singular event log

### PR DESCRIPTION
**Pull request by kenanmamedoff**

### CODE REVIEW

File: brd-ios/breadwallet/Helpers/ApplicationController.swift [hunk 1]
```diff
@@ -115,9 +111,28 @@ class ApplicationController: Subscriber {
...
-        return SingularConfig(apiKey: E.singularApiKey,
-                              andSecret: E.singularSecretKey)
+    private func getSingularConfig() -> SingularConfig? {
+        let config = SingularConfig(apiKey: E.singularApiKey,
+                                    andSecret: E.singularSecretKey)
+        return config
+    }
+    
+    private func setupSingular() {
+        var config = getSingularConfig()
```
Comment (dgacevic) [inline brd-ios/breadwallet/Helpers/ApplicationController.swift:121]: Separate function for this simple init seems redundant

File: brd-ios/breadwallet/Helpers/ApplicationController.swift [hunk 2]
```diff
@@ -115,9 +111,28 @@ class ApplicationController: Subscriber {
...
+        let kycStatus = String(describing: UserManager.shared.profile?.status)
+        let authKeyString = Backend.apiClient.authKey?.encodeAsPublic.hexToData?.base58 ?? ""
+        let os = UIDevice.current.systemName + " " + UIDevice.current.systemVersion
+        let version = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String
+        let build = Bundle.main.infoDictionary?["CFBundleVersion"] as? String
+        let globalProperties: [String: String] = ["Device": "\(UIDevice.current.name)",
```
Comment (dgacevic) [inline brd-ios/breadwallet/Helpers/ApplicationController.swift:121]: Should theses keys be added to constants?

File: brd-ios/breadwallet/Helpers/SingularAnalytics.swift [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import Foundation
+import Singular
+
+protocol SingularAnalyticsLogData {
+    var name: String { get set }
+    var data: [String: String] { get set }
+}
+
+struct SingularAnalytics {
+    static var shared = SingularAnalytics()
```
Comment (dgacevic) [inline brd-ios/breadwallet/Helpers/SingularAnalytics.swift:20]: Singleton struct?

File: brd-ios/breadwallet/Helpers/ApplicationController.swift (hunk 2 above)
Comment (kenanmamedoff) [reply brd-ios/breadwallet/Helpers/ApplicationController.swift:121]: may be. i am not really sure. these keep changing all the time and only being used in 1 place. let's leave it for now.

File: brd-ios/breadwallet/Helpers/SingularAnalytics.swift (hunk 3 above)
Comment (kenanmamedoff) [reply brd-ios/breadwallet/Helpers/SingularAnalytics.swift]: that was nonsense. removed.

**Review by dgacevic — DISMISSED** [review]

**Review by dgacevic — APPROVED** [review]

**Pull request merged**